packages/extension-commerce-bm-seo/src/components/seo-hoc.tsx

```diff
@@ -71,10 +97,14 @@ const seoHOC = <P extends object>(WrappedComponent: React.ComponentType<P>) => {
                 }
             }
             void fetchData().catch(console.error)
-        }, [urlSegment])
+        }, [urlSegment, skipMappingCall])
 
         const {isBlocked: isNavigationBlocked} = useBlockNavigation(
             async (location: Location, _: string) => {
+                if (skipMappingCall) {
+                    return
+                }
```

Comment:
Nice.. lets at some inline comment here stating we are making an early exit because of the current configuration.